catalog/transformer/catalog-transformer-pptx/src/test/java/ddf/catalog/transformer/input/pptx/PptxInputTransformerTest.java

```diff
@@ -0,0 +1,224 @@
+            "/ae.ac.uaeu.faculty_nafaachbili_GeomLec1.pptx",
+            "/alterman_security.pptx",
+            "/at.ecodesign.www_downloads_Vertiefungsvortrag_elektronik.pptx",
+            "/backgrounds.pptx",
+            "/layouts.pptx",
+            "/pie-chart.pptx",
+            "/pptx2svg.pptx",
+            "/prProps.pptx",
+            "/present1.pptx",
+            "/sample.pptx",
+            "/sample_pptx_grouping_issues.pptx",
+            "/shapes.pptx",
+            "/table_test.pptx",
+            "/testPPT.pptx",
+            "/themes.pptx",
+            "/with_japanese.pptx",
+            "/hello-world-signed.pptx",
+            "/hello-world-unsigned.pptx",
+            "/ms-office-2010-signed.pptx");
+
+    private final InputTransformer inputTransformer = new TikaInputTransformer(null);
+
+    private InputStream getResource(String resourceName) {
+        return PptxInputTransformerTest.class.getResourceAsStream(resourceName);
+    }
+
+    @Test(expected = CatalogTransformerException.class)
+    public void testBadCopy() throws IOException, CatalogTransformerException {
+        try {
+            PptxInputTransformer t = new PptxInputTransformer(inputTransformer);
+            InputStream is = Mockito.mock(InputStream.class);
+            Mockito.when(is.read(Mockito.any()))
+                    .thenThrow(new IOException());
+            t.transform(is);
+        } catch (CatalogTransformerException e) {
+            Assert.assertTrue(e.getMessage()
+                    .contains("Could not copy bytes of content message."));
+            throw e;
+        }
+    }
+
+    @Test(expected = CatalogTransformerException.class)
+    public void testTransformNullInput() throws IOException, CatalogTransformerException {
+        PptxInputTransformer t = new PptxInputTransformer(inputTransformer);
+        t.transform(null);
+    }
+
+    @Test
+    public void testBunchOfPptxFiles() throws IOException, CatalogTransformerException {
+        for (String file : files) {
+            PptxInputTransformer t = new PptxInputTransformer(inputTransformer);
+            try (InputStream is = getResource(file)) {
+                Metacard mc = t.transform(is);
+                assertNotNull(mc.getThumbnail());
+            }
+        }
+    }
+
+    @Test(expected = CatalogTransformerException.class)
+    public void testPasswordProtected() throws IOException, CatalogTransformerException {
+        PptxInputTransformer t = new PptxInputTransformer(inputTransformer);
+        try (InputStream is = getResource("/password-powerpoint.pptx")) {
+            t.transform(is);
+        }
+    }
+
+    @Test
+    public void testOle2() throws IOException, CatalogTransformerException {
+        PptxInputTransformer t = new PptxInputTransformer(inputTransformer);
+        try (InputStream is = getResource("/empty.ppt")) {
+            Metacard mc = t.transform(is);
+            assertEquals(null, mc.getThumbnail());
```

Comment:
Generally, `assertThat()` is the preferred assert method. [This link](https://objectpartners.com/2013/09/18/the-benefits-of-using-assertthat-over-other-assert-methods-in-unit-tests/) points out its benefits.